lib/src/model.dart

```diff
@@ -4841,59 +4985,233 @@ abstract class LibraryContainer extends Nameable
   /// that only represent a part of the SDK.
   bool get isSdk => false;
 
-  /// Order by which this container should be sorted.
-  List<String> get containerOrder;
-
   /// Returns:
   /// -1 if this container is listed in [containerOrder].
   /// 0 if this container is named the same as the [enclosingName].
   /// 1 if this container represents the SDK.
   /// 2 if this group has a name that contains the name [enclosingName].
   /// 3 otherwise.
   int get _group {
-    if (containerOrder.contains(name)) return -1;
-    if (equalsIgnoreAsciiCase(name, enclosingName)) return 0;
+    if (containerOrder.contains(sortKey)) return -1;
+    if (equalsIgnoreAsciiCase(sortKey, enclosingName)) return 0;
     if (isSdk) return 1;
-    if (name.toLowerCase().contains(enclosingName.toLowerCase())) return 2;
+    if (sortKey.toLowerCase().contains(enclosingName.toLowerCase())) return 2;
     return 3;
   }
 
   @override
   int compareTo(LibraryContainer other) {
     if (_group == other._group) {
       if (_group == -1) {
-        return Comparable.compare(
-            containerOrder.indexOf(name), containerOrder.indexOf(other.name));
+        return Comparable.compare(containerOrder.indexOf(sortKey),
+            containerOrder.indexOf(other.sortKey));
       } else {
-        return name.toLowerCase().compareTo(other.name.toLowerCase());
+        return sortKey.toLowerCase().compareTo(other.sortKey.toLowerCase());
       }
     }
     return Comparable.compare(_group, other._group);
   }
 }
 
-/// A category is a subcategory of a package, containing libraries tagged
-/// with a @category identifier.  Comparable so it can be sorted according to
-/// [DartdocOptionContext.categoryOrder].
-class Category extends LibraryContainer {
-  final String _name;
+abstract class MarkdownFileDocumentation
+    implements Documentable, Canonicalization {
+  DocumentLocation get documentedWhere;
+
+  @override
+  String get documentation => documentationFile?.contents;
+
+  Documentation __documentation;
+  Documentation get _documentation {
+    if (__documentation != null) return __documentation;
+    __documentation = new Documentation.forElement(this);
+    return __documentation;
+  }
+
+  @override
+  String get documentationAsHtml => _documentation.asHtml;
+
+  @override
+  bool get hasDocumentation =>
+      documentationFile != null && documentationFile.contents.isNotEmpty;
+
+  @override
+  bool get hasExtendedDocumentation =>
+      documentation != null && documentation.isNotEmpty;
+
+  @override
+  bool get isDocumented;
+
+  @override
+  String get oneLineDoc => __documentation.asOneLiner;
+
+  FileContents get documentationFile;
 
+  // TODO: implement location
+  @override
+  String get location => pathLib.toUri(documentationFile.file.path).toString();
+
+  // TODO: implement locationPieces
```

Comment:
ditto

Reply:
actually forgot to implement this.  That will fail an assert elsewhere if we ever warn on a subclass of this, so implemented